lib/escript_testbed/scenario.ex

```diff
@@ -1,8 +1,14 @@
-defmodule EscriptTestbed.Scenario do
-  @moduledoc """
-  Common behaviour for all scenarios.
-  """
+# defmodule EscriptTestbed.Scenario do
+#   @moduledoc """
+#   Common behaviour for all scenarios.
+#   """
 
-  @doc "Invokes the sync scenario execution"
-  @callback run() :: any()
+#   @doc "Invokes the sync scenario execution"
+#   @callback run() :: any()
+# end
+
+defprotocol EscriptTestbed.Scenario do
```

Comment:
Found another way to do the same without touching the scenarios themselves (no need for that `struct` trick). Just implement protocol for `Atom`:

```
defimpl EscriptTestbed.Scenario, for: Atom do
  def run(EscriptTestbed.Scenarios.DataSetA) do
    EscriptTestbed.Scenarios.DataSetA.run()
  end

  def run(EscriptTestbed.Scenarios.DataSetB) do
    EscriptTestbed.Scenarios.DataSetB.run()
  end
end
```

And in CLI:
```
  defp cli_param_to_scenario(term) do
    module_suffix = Macro.camelize(term) |> String.to_atom()
    Module.safe_concat([EscriptTestbed.Scenarios, module_suffix])
  end
```

Achieves the same, and at least the protocol trick is hidden within a single file.